### ISSUE TITLE
VX30 Weightings Are Significantly Off VXX Current Holdings

### ISSUE THREAD

**Issue by UpsidePotential** [dougransom/vix_utils#30]
Since VXX follows the VX30 index, I would expect he holdings to be close to the calculated VX30 weightings.   
![image](https://github.com/dougransom/vix_utils/assets/48447427/85c7e0b9-7ef9-4b87-b7bf-b840b20cc3f3)

![image](https://github.com/dougransom/vix_utils/assets/48447427/6bc8fedc-0a5d-4366-9175-7f7fa773a75d)

**Comment by githubbla** [comment]
did you able to download vix-utils features, it's looks like M2 broken for now.

**Comment by gnzsnz** [comment]
weights are definitively off

```python
import vix_utils as vu

vix_dates = vu.vix_futures_dates.vix_futures_trade_dates_and_expiry_dates()
weights = vu.vix_futures_dates.vix_constant_maturity_weights(vix_dates)
```
![image](https://github.com/dougransom/vix_utils/assets/8376642/ad7520e0-567f-459a-a316-a0a149bf704b)

**Comment by dougransom** [comment]
I am just starting to look at this now.  Anyone already looking at it?

**Comment by dougransom** [comment]
Roll Period Calendar Days is Zero anywhere thse weights are off.  I have some unit tests I am working on to identify these cases.

**Comment by gnzsnz** [comment]
there are many cases for 2024

![image](https://github.com/dougransom/vix_utils/assets/8376642/9c8951b7-55ef-4a52-9c02-8784310f1264)

the issue should be around this line of code (before it actually)

https://github.com/dougransom/vix_utils/blob/1b231b2c7fc44ae84584496940ac068139a2e55b/src/vix_utils/vix_futures_dates.py#L184

but i have not been able to pin point it

I don't think that roll period calendar days is the main issue. There are plenty of good weights with it being zero

![image](https://github.com/dougransom/vix_utils/assets/8376642/bedab64e-94d1-41ac-8bee-1168a5e17d18)

**Comment by dougransom** [comment]
Working on this in branch https://github.com/dougransom/vix_utils/tree/weights.  

So far some refactorings, so we can debug the logic that generates the weights on a small subset of dates.

**Comment by dougransom** [comment]
The orginal code was far off the documentation.  I think I am close to resolving …

**Comment by dougransom** [comment]
Closer still, it might a week or two though before I complete it.  There is an edge case on Wednesday of Vix  Future Expiration.  That is the front month (at least I think it is), but on those days, for the purpose of  S&P 500 VIX Short-Term Futures Index, the front month is the following expiration.

**Comment by dougransom** [comment]
I think it is working now, try `vixutil -w w.xlsx`.

Note the weights are for three expirations, since there are a couple days where the front month may not be included in VXX etc. (instead it is the following two months).  It happens the day before and day of expiration.

**Comment by dougransom** [comment]
The latest on PyPi has this fix now 0.1.7.

**Comment by gnzsnz** [comment]
thanks a lot!

so now front month goes to zero the friday before expiration. it makes sense to me.
